public/app/features/alerting/unified/RuleEditor.test.tsx

```diff
@@ -0,0 +1,78 @@
+import { Route } from 'react-router';
+import { selectOptionInTest } from 'test/helpers/selectOptionInTest';
+import { render, screen, userEvent } from 'test/test-utils';
+
+import { setAppEvents } from '@grafana/runtime';
+import appEvents from 'app/core/app_events';
+import RuleEditor from 'app/features/alerting/unified/RuleEditor';
+import { setupMswServer } from 'app/features/alerting/unified/mockApi';
+import { grantUserPermissions, mockDataSource } from 'app/features/alerting/unified/mocks';
+import { setupDataSources } from 'app/features/alerting/unified/testSetup/datasources';
+import { DataSourceType } from 'app/features/alerting/unified/utils/datasource';
+import { AccessControlAction } from 'app/types';
+
+setupMswServer();
+
+// Required to make sure that loading the datasource plugin does not fail
+// see public/app/plugins/datasource/loki/module.ts
+setAppEvents(appEvents);
```

Comment:
This logic is run in `app.ts`, and should probably be run before all of our tests, as parts of the code rely on it having been setup (otherwise the attempted `subscribe` method in the linked file would fail)